### PR TITLE
shardddl: fix resume conflict ddl.

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
#719 #582 

### What is changed and how it works?
- reset shardmeta after start or resume task
- mark `ErrSyncerShardDDLConflict`  as UnresumableError

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: syncer/sharding-meta/shardmeta.go [hunk 1]
```diff
@@ -178,25 +178,23 @@ func (meta *ShardingMeta) checkItemExists(item *DDLItem) (int, bool) {
 //   active: whether the DDL will be processed in this round
 func (meta *ShardingMeta) AddItem(item *DDLItem) (active bool, err error) {
 	index, exists := meta.checkItemExists(item)
-	if exists {
-		return index == meta.activeIdx, nil
```
Comment (csuzhangxc) [inline syncer/sharding-meta/shardmeta.go:182]: Is it easy to add a negative test case for #719?

File: tests/sequence_sharding/run.sh [hunk 2]
```diff
@@ -36,6 +36,38 @@ function run() {
...
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "query-status sequence_sharding" \
+        "detect inconsistent DDL sequence" 2
+
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "stop-task sequence_sharding" \
+        "\"result\": true" 3
+    
+    cp $cur/conf/dm-task.yaml $WORK_DIR/task.yaml
+    echo "ignore-checking-items: [\"all\"]" >> $WORK_DIR/task.yaml
```
Comment (WangXiangUSTC) [inline tests/sequence_sharding/run.sh:62]: why need to ignore all checks? how about adding a comment

Comment (GMHDBJD) [reply tests/sequence_sharding/run.sh:63]: add comment in 644a504

**Comment by GMHDBJD** [comment]
> rest LGTM, and which modify related to the description `check ddl conflict even the ddl exist`?

We have a discuss in https://github.com/pingcap/dm/issues/582#issuecomment-640321282 .
Now we reset the shardmeta when resume and it will check conflict again. The description `check ddl conflict even the ddl exist`  is outdated now, I will update it.

**Comment by csuzhangxc** [comment]
@GMHDBJD please cherry-pick it manually again. 😢

**Pull request merged**